### PR TITLE
Fixed bug #1849

### PR DESCRIPTION
**Pull request by crisdut**
Resolution:
- Add parameter sourceTimeZone in the method ConvertTimeToUtc();

https://bugzilla.xamarin.com/show_bug.cgi?id=1849

### CODE REVIEW

**Comment by tritao** [comment]
@esdrubal can you review this?

**Comment by esdrubal** [comment]
build

File: mcs/class/System.Core/Test/System/TimeZoneInfoTest.cs [hunk 1]
```diff
@@ -519,6 +519,34 @@ public void ConvertTime_AdjustmentRuleAfterNewYears ()
 				Assert.AreEqual (0, ddt.Minute, "#3.5");
 				Assert.AreEqual (0, ddt.Second, "#3.6");
 			}
+
+			[Test (Description="Fix the bug https://bugzilla.xamarin.com/show_bug.cgi?id=1849")]
+			public void ConvertTime_AjustmentConvertTimeWithSourceTiemZone() {
```
Comment (esdrubal) [inline mcs/class/System.Core/Test/System/TimeZoneInfoTest.cs]: Typo "TiemZone"

**Comment by esdrubal** [comment]
Thanks for the pull request.
Squashed last two commits before merging in 7c39ab701bcee11156cbf505587bc2d630bf819c.

**Pull request closed without merging**